app/models/vm_or_template.rb

```diff
@@ -51,7 +51,7 @@ class VmOrTemplate < ApplicationRecord
   validates_presence_of     :name, :location
   validates_inclusion_of    :vendor, :in => VENDOR_TYPES.values
 
-  has_one                   :miq_server, :foreign_key => :vm_id
+  has_one                   :miq_server, :foreign_key => :vm_id, :inverse_of => :vm
```

Comment:
`MiqServer belongs_to :vm`
When we ask for `vm.miq_server` I wanted the `vm` to be assigned.

There is no way to assign the miq_server to the `:vm` and the `:template` and all those others.

The usecase is `template.miq_server.vm`. Is this valid?